<launch>
    <arg name="pcd_folder" default="/apollo/data/pcd"/>
    <arg name="skip_static_frames" default="false"/>
    <arg name="child_frame_id" default="velodyne64"/>
    <arg name="stamp_file" default="$(arg pcd_folder)/stamp.txt"/>
    <arg name="pose_file" default="$(arg pcd_folder)/pose.txt"/>
    <arg name="topic_pointcloud" default="/apollo/sensor/velodyne64/PointCloud2"/>
    <arg name="queue_size" default="10"/>
    <arg name="nodelet_manager_name" default="velodyne_nodelet_manager" />

    <node pkg="nodelet" type="nodelet" name="pcd_exporter_nodelet"
                args="load velodyne_pointcloud/PCDExporterNodelet $(arg nodelet_manager_name)">
        <param name="pcd_folder" value="$(arg pcd_folder)"/>
        <param name="skip_static_frames" value="$(arg skip_static_frames)"/>
        <param name="child_frame_id" value="$(arg child_frame_id)"/>
        <param name="stamp_file" value="$(arg stamp_file)"/>
        <param name="pose_file" value="$(arg pose_file)"/>
        <param name="topic_pointcloud" value="$(arg topic_pointcloud)"/>
        <param name="queue_size" value="$(arg queue_size)"/>
    </node>
</launch>
